machine/clients/slack.py

```diff
@@ -93,6 +95,80 @@ async def _process_users_channels(self, _: AsyncBaseSocketModeClient, req: Socke
             elif event["type"] == "member_joined_channel":
                 await self._on_member_joined_channel(event)
 
+    async def build_paginated_cache(
```

Comment:
Have you considered making a generator out of this so that you don't have to pass the register method into it. Instead, you would call the generator with the client method and data key and pass each yielded value into the register method.
That feels a bit more "functional" to me.

Reply:
I updated the code so that it is a generator instead. Thanks for the tip.